### PR TITLE
client: add gethostbyname6, add ipv6 fallback to ohostname

### PR DESCRIPTION
**Pull request by cfcs**
client interface:
  - add gethostbyname6
  - add ipv6 fallback to ohostname
  - add ipv6 AAAA in odns

### CODE REVIEW

File: app/odns.ml [hunk 1]
```diff
@@ -70,6 +71,16 @@ let for_all_domains nameserver ~domains typ f =
...
   | () -> Ok () (* TODO catch failed jobs *)
 
+let do_aaaa nameserver domains _ =
+  for_all_domains nameserver ~domains Udns.Rr_map.Aaaa
+    (fun domain -> function
+       | Ok aaaa_resp ->
+         Logs.app (fun m -> m "%a" pp_zone
+                      (domain, Udns.Rr_map.Aaaa, aaaa_resp))
+       | Error (`Msg msg) ->
+         Logs.err (fun m -> m "Failed to lookup %a: %s\n%!"
```
Comment (hannesm) [inline app/odns.ml]: I'd embed the lookup type (i.e. AAAA) in the error log as well.

File: unix/client/ohostname.ml [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
   let t = Udns_client_unix.create () in
+  let domain = Domain_name.of_string_exn Sys.argv.(1) in
   let res =
-    Udns_client_unix.gethostbyname t (Domain_name.of_string_exn Sys.argv.(1)) in
+    Udns_client_unix.gethostbyname t domain in
   match res with
   | Ok addr -> Fmt.pr "%a\n" Ipaddr.V4.pp addr
-  | Error (`Msg x) -> Fmt.epr "Failed to resolve: %s\n" x; exit 1
+  | Error _ ->
+    (* Here we fall back to trying to resolve the IPv6 address instead: *)
```
Comment (hannesm) [inline unix/client/ohostname.ml]: looks good, but: should `ohostname` be similar to the unix `host` utility? if so, actually both A and AAAA queries should be performed (as well as a MX lookup)

Comment (cfcs) [reply unix/client/ohostname.ml]: That was indeed the intention. Maybe it should also be renamed to `ohost` ? I always get those two mixed up...

Good point, that also seems useful, I'll do that.

Comment (hannesm) [reply unix/client/ohostname.ml]: sounds good to call it `ohost`.

File: app/odns.ml [hunk 3]
```diff
@@ -66,32 +67,47 @@ let for_all_domains nameserver ~domains typ f =
...
+                         m "Failed to lookup %a for %a: %s\n%!"
+                           Udns_enum.pp_rr_typ (Udns.Rr_map.k_to_rr_typ typ)
+                           Domain_name.pp domain msg) ;
+                     res)
                 >|= f domain)
              domains) with
   | () -> Ok () (* TODO catch failed jobs *)
 
+let pp_response typ domain = function
+  | Error _ -> ()
```
Comment (hannesm) [inline app/odns.ml]: hnm, now there's no more output when an error (lookup failure) is encountered... is this intentional?

Comment (hannesm) [reply app/odns.ml]: > am confused by is the WARNING stuff that appears above the error

this is part of the [convenience function `is_reply`](https://github.com/roburio/udns/blob/aeee951f487cf8aa9b20720cc4860a9c3fa24b70/src/udns.ml#L2560) when invoked without `~not_error:false`. Maybe the `rcode` checking should not be done in there if this is confusing (I had to disable that in the server components as well)... maybe `Packet.is_reply` should do less work, not log, but instead return a result...

**Pull request merged**